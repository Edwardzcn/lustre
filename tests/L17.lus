type S = { x : int; y : int };

node main(s : S; x : int) returns (t : S)
let
  t = s { x := x };
tel
