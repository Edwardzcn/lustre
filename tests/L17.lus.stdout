PhaseToCore
===========

node (s.x : int, s.y : int, x : int)
  returns (t.x, t.y)
  assumes ()
  shows ()
let
  t.x : int
    = x
   
  t.y : int
    = s.y
tel

--- Step 1 ---
s.x : int  = 1
s.y : int  = 2
x : int  = 3
t.x = 3
t.y = 2
--- Step 2 ---
s.x : int  = 19
s.y : int  = 20
x : int  = 21
t.x = 21
t.y = 20
--- Step 3 ---
s.x : int  = (EOF)
